### ISSUE TITLE
non-Sendable captures don't suggest the most common solution 

### ISSUE THREAD

**Issue by mattmassicotte** [apple/swift#74388]
### Motivation

Here's the example I'm thinking of:

```swift
@MainActor
class MyClass {
}

func someFunction(_ block: @Sendable () -> Void) {
}

func useFunction() {
    let value = MyClass()

    someFunction {
        // ERROR: Capture of 'value' with non-sendable type 'MyClass' in a `@Sendable` closure
        // Info: Class 'MyClass' does not conform to the 'Sendable' protocol
        print(value)
    }
}
```

This code produces an error and a extra bit of info. One of the most common solutions to a problem like this is to use a global actor. But users need to known that's an option and that it will make the type Sendable. I think the compiler should offer that advice!

### Proposed solution

How about something like:

```
// ERROR: Capture of 'value' with non-sendable type 'MyClass' in a `@Sendable` closure
// Info: Class 'MyClass' does not conform to the 'Sendable' protocol
// Info: Class 'MyClass' is not isolated to a global actor
```

**Comment by wadetregaskis** [comment]
I don't understand your proposed amendment; `MyClass` _is_ isolated to a global actor (`@MainActor`)…?

A better fix-it suggestion might be to replace the `Sendable` requirement with `sending`, as in this case specific example that's all that's needed to type check successfully in `useFunction`.

I'm also not sure that the global actor isolation is that relevant, anyway.  Irrespective of whether a type is isolated to a global actor, it can still be _sent_ to other isolation domains (IIRC), it just can't be _used_ from them.  The crux really is the sending aspect, and the ambiguity is the user's intent as to whether they actually want to send it or not.

**Comment by mattmassicotte** [comment]
Arg, I typo'ed. Yes, that `@MainActor` is not supposed to be there in the example. I've just removed it. And, yes once actor-isolated, that type is indeed Sendable. So adding that `@MainActor` make the above code valid.

I love the *concept* of mentioning `sending` here as well. I just don't know enough about it to suggest it with confidence. I can say, though, that often when faced with this kind of thing, the closure in question isn't under your control. I'm not sure if such a suggestion could be conditionalized on the definition being within the same module maybe?

**Comment by wadetregaskis** [comment]
Yeah, I'm not sure _entirely_ what it would mean to say a _closure_ is `sending` (I mean, literally it would mean that anything it captures must be carried with it, for isolation purposes, but the question is can that actually be implemented in the compiler and does it have any undesirable logical consequences).  Anyway, that's pretty tangential now that you've corrected your example.
